### PR TITLE
Move string-format to sol_buffer usage

### PR DESCRIPTION
**Pull request by glima**

### CODE REVIEW

File: src/lib/datatypes/sol-buffer.c [hunk 1]
```diff
@@ -133,6 +133,35 @@ sol_buffer_append_slice(struct sol_buffer *buf, const struct sol_str_slice slice
...
+{
+    int err;
+
+    SOL_NULL_CHECK(buf, -EINVAL);
+    if (buf->used < pos) {
+        return -EINVAL;
+    }
+
+    /* Extra room for the ending NUL-byte. */
+    if (slice.len >= SIZE_MAX - 1 - pos)
```
Comment (lpereira) [inline src/lib/datatypes/sol-buffer.c]: Use nul_byte_size() here because of the "no nul byte" flag.

Comment (glima) [inline src/lib/datatypes/sol-buffer.c]: Ouch, true! Will fix.

**Comment by glima** [comment]
Thanks, @lpereira  and @edersondisouza . If no more issues are pointed till the end of the day, I'll fix those and merge.

**Comment by glima** [comment]
Merged.

**Pull request closed without merging**